app/api/helpers.py

```diff
@@ -6,16 +6,34 @@
 from os import urandom
 from binascii import hexlify
 from flask import abort as flask_abort
-from flask import current_app, jsonify, make_response, render_template_string
-from flask_login import current_user
+from flask import current_app, jsonify, make_response, render_template_string, request
+from flask_login import current_user, login_user
 from werkzeug.exceptions import HTTPException
 
 from app.models import Agency, BriefUser, User, db
 from app.tasks.email import send_email
+from app.authentication import get_api_key_from_request
 from dmutils.csrf import get_csrf_token
 from dmutils.email import ONE_DAY_IN_SECONDS, EmailError, parse_fernet_timestamp
 
 
+def allow_api_key_auth(func):
+    @wraps(func)
+    def decorated_view(*args, **kwargs):
+        request_key = get_api_key_from_request(request)
+        if request_key:
+            from app.api.services.api_key import ApiKeyService
+            api_key_service = ApiKeyService()
+            key = api_key_service.get_key(request_key)
+            if not key:
+                return flask_abort(403, 'Invalid API key - revoked or non existent')
+            login_user(key.user)
+            current_app.logger.info('login.api_key.success: {user}', extra={'user': key.user.name})
+            return func(*args, **kwargs)
```

Comment:
Do we need to `return` here since it can return when it breaks out of the `if` block?